normandy/recipes/forms.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+from django import forms
+from django.core.exceptions import ValidationError
+
+import jsonschema
+
+
+class ActionAdminForm(forms.ModelForm):
+    def clean(self):
+        """
+        Validate that the action being edited isn't currently in use.
+        """
+        super().clean()
+
+        if self.instance.in_use:
+            self.add_error(
+                None,
+                ValidationError('Action is currently in use and cannot be edited.')
+            )
+
+
+class RecipeActionInlineForm(forms.ModelForm):
+    def clean(self):
+        """Validate the arguments against their schema."""
```

Comment:
This is definitely an open question when mixing forms and DRF. Kicking it down the road it is!